### PR TITLE
Domains: Add professional email upsell route

### PR DESCRIPTION
**Pull request by runofthemillgeek**
#### Changes proposed in this Pull Request

This PR is the first step to proposed changes in domain purchase flow. It adds a new route for the professional email upsell page at:

```
http://wordpress.com/domains/add/:domain/email/:siteName
```

The content for this page will be added in an upcoming PR based on this one.

Side note: I referred to @daledupreez's previous approach at #50152 and @stephanethomas's work on #53712 to come up with this.

#### Testing instructions

1. Run git checkout `add/titan-email-upsell-route` and start your server
2. Log into a WordPress.com account that has a site
3. Open the _Domains_ page (replace `yoursite` with a site slug that you have): `http://calypso.localhost:3000/domains/manage/yoursite.wordpress.com`
4. Click Add _Domain button > to this site_ from the popover
5. Search for a domain or choose one of the suggestions by clicking _Select button_
5. Once the page loads, replace `google-workspace` in the domain with `email`. For example:
    If the URL shown is:
    ```
    http://calypso.localhost:3000/domains/add/example.in/google-workspace/example.wordpress.com
    ```
   
    replace it with:

    ```
    http://calypso.localhost:3000/domains/add/example.in/email/example.wordpress.com
    ```
6. Verify that the new route loads and renders without errors

#### Screenshots

GOwwpLwluM

### CODE REVIEW

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx]: I think it would be better to use `domainAddNew( selectedSiteSlug)` from `calypso/my-sites/domains/paths.js`.

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx]: Nit: The Letero team has been trying to use `<>` instead of `Fragment` or `React.Fragment`.
Not blocking, but a minor thing.

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx:41]: As a question, is the goal to improve/expand this content in a follow-up PR?

File: client/my-sites/domains/controller.jsx
Comment (daledupreez) [inline client/my-sites/domains/controller.jsx:23]: Nit: I know this import list isn't in order, but I'd prefer that the new import not make it less ordered! Could you move this new import one line up so it's below `import DomainSearch`?

File: client/my-sites/domains/controller.jsx
Comment (daledupreez) [inline client/my-sites/domains/controller.jsx]: For full specificity and clarity, it may be better to call this `emailUpsellForDomainRegistration`.

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx]: Given that we're only using `translateArgs` once, it may be simpler to inline the details.

**Review by daledupreez — COMMENTED** [review]
This is looking really good, @sangeeth96! The description and testing instructions covered pretty much everything. The only thing that may be worth adding is making it clear that the upsell content will be coming in another PR.

Other than that, I have a few small comments that you can take on. I am hoping we can get this shipped tomorrow!

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (runofthemillgeek) [reply client/my-sites/domains/email-providers-upsell/index.jsx]: @daledupreez I used that first, but then thought you peeps prefer to be explicit about it and changed it back anyway. 😅  But will do this. ✌️

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (runofthemillgeek) [reply client/my-sites/domains/email-providers-upsell/index.jsx:41]: @daledupreez Yes, that is indeed the goal. I've updated the PR description as you requested.

File: client/my-sites/domains/controller.jsx
Comment (runofthemillgeek) [reply client/my-sites/domains/controller.jsx:23]: On it. Is there an eslint config or some other editor automation folks use to automate this or is this a manual work at the moment? 🤔

Comment (runofthemillgeek) [reply client/my-sites/domains/controller.jsx:23]: Oh nvm, I see that work is being done on this. ❤️

**Comment by runofthemillgeek** [comment]
@daledupreez Addressed the comments. Let me know if there's more.

File: client/my-sites/domains/controller.jsx
Comment (daledupreez) [inline client/my-sites/domains/controller.jsx]: Small bug: this should be `context.params.domain` - `registerDomain` is not defined in the main URL path as a variable.
```suggestion
					args: { domain: context.params.domain },
```

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx]: This should be under Internal dependencies and should be relative to the repo rather than the current file. (I''ll add a comment RE where to add.)
```suggestion
```

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (daledupreez) [inline client/my-sites/domains/email-providers-upsell/index.jsx:13]: Location to import `domainAddNew`.
```suggestion
import { domainAddNew } from 'calypso/my-sites/domains/paths';
import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
```

**Review by daledupreez — COMMENTED** [review]
I should have gotten to this earlier today! 😐 

There are still two more fixes/tweaks I'd like you to add before we merge and deploy. (I was going to try and handle them myself, but I don't have permissions to push them to the `talk-to` repo.)

File: client/my-sites/domains/controller.jsx
Comment (runofthemillgeek) [reply client/my-sites/domains/controller.jsx]: Whoops, gonna fix that.

Comment (runofthemillgeek) [reply client/my-sites/domains/controller.jsx]: Done.

File: client/my-sites/domains/email-providers-upsell/index.jsx
Comment (runofthemillgeek) [reply client/my-sites/domains/email-providers-upsell/index.jsx]: Made the change to use the repo alias.

**Comment by runofthemillgeek** [comment]
> I don't have permissions to push them to the `talk-to` repo.)

I genuinely thought PRs were editable by upstream maintainers but turns out org forks can't allow this (isaacs/github#1681).

I'll see if I can add you to the repo's collaborators list.

**Review by daledupreez — APPROVED** [review]
LGTM! I'll get this deployed first thing tomorrow.

Congrats on your first Calypso PR! 🎉

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6303536

Thank you @sangeeth96 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6303536).

**Pull request merged**
